codecombat/codecombat
# Add UTF-8 support for cpp

# Issue

The browser `atob` function doesn't work with UTF-8 [[reference](https://developer.mozilla.org/en-US/docs/Web/API/WindowBase64/Base64_encoding_and_decoding#The_Unicode_Problem)].

This leads to utf-8 strings in the users code to become destroyed.

For example, the following `hero.say("abӁcd");` becomes `hero.say("abÓ cd");` if using the native `atob` function.

# Fix

The easiest fix is to use a library.

# Manual test

1. As an admin start a local environment and use `npm run proxy` to connect to production server.
2. Choose C++ language
3. Add the code `hero.say("abӁcd");` into a level and check that the hero actually says that.
